drivers/pwm/pwm_max31790.c

```diff
@@ -0,0 +1,466 @@
+/*
+ * Copyright (c) 2023 SILA Embedded Solutions GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT maxim_max31790
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/pwm.h>
+#include <zephyr/drivers/pwm/max31790.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(pwm_max31790, CONFIG_PWM_LOG_LEVEL);
+
+#define MAX31790_OSCILLATOR_FREQUENCY_IN_HZ 32768
+#define MAX31790_PWMTARGETDUTYCYCLE_MAXIMUM ((1 << 9) - 1)
+#define MAX31790_TACHTARGETCOUNT_MAXIMUM    ((1 << 11) - 1)
```

Comment:
Got your point, but I am aiming here not for a mask, but instead for the maximum possible value. Therefore it makes more sense in my opinion to calculate it like this, although it would be the same result as with GENMASK.